footil/date.py

```diff
@@ -8,83 +8,116 @@
 
 
 def _modify_date(
-    dt: str or datetime.date,
-    fmt: str,
+    dt: datetime.date,
     modify_fun: callable,
-    res_type: Optional[str] = 'str',
-    new_fmt: Optional[str] = False,
     args: Optional[tuple] = None,
-        kwargs: Optional[dict] = None) -> str:
+        kwargs: Optional[dict] = None) -> datetime.date:
     if not args:
         args = []
     if not kwargs:
         kwargs = {}
-    if not new_fmt:
-        new_fmt = fmt
-    if type(dt) == str:
-        dt = datetime.datetime.strptime(dt, fmt).date()
-    modified_dt = modify_fun(dt, *args, **kwargs)
-    if res_type == 'date':
-        return modified_dt
-    else:
-        return modified_dt.strftime(new_fmt)
+    return modify_fun(dt, *args, **kwargs)
+
+
+def _convert_date_str_to_date(
```

Comment:
we can simplify function names (like in Odoo), to_string, to_date. This is date module, so it is kind of clear what it does anyway.